### DISCUSSION TITLE
would like to block specific ip addresses and/or ip address ranges

### DISCUSSION THREAD

**Discussion by roblatour** [homeassistant-apps/app-cloudflared#331] [Ideas]
Every so often I get an alert in Home Assistant saying "Login attempt or request with invalid authentication from  specific IP address (specific IP address). See the log for details.

These are not being made by me, rather I strongly suspect they are being made by bots.  

It would be great if on the Home Assistant Cloudflare config setting window,  or in some other manner, that I could specify specific ip addresses and/or ip address ranges to be blocked.

**Comment by brenner-tobias** [comment]
You can use Network policies in Cloudflare Zero Trust for that, as described [here](https://developers.cloudflare.com/cloudflare-one/policies/filtering/network-policies/).

**Reply by roblatour** [reply]
@brenner-tobias thank you so much for this.  I am using the free cloud service, your post got me looking deeper and indeed I can address my needs with the Security - Custom rules - WAF (Web Application Firewall) associated with it.
